bridge/client/webrtc.js

```diff
@@ -1365,32 +1367,60 @@
             });
         });
 
+        function getDataLength(data) {
+            if (data instanceof Blob)
+                return data.size;
+
+            if (data instanceof ArrayBuffer || ArrayBuffer.isView(data))
+                return (new Uint8Array(data)).byteLength;
+
+            return unescape(encodeURIComponent(data)).length;
+        }
+
         function encodeArrayBufferArgument(buffer) {
-            var binary = '';
+            var binary = "";
             var bufferView = new Uint8Array(buffer);
-            for (var i = 0, len = bufferView.byteLength; i < len; i++) {
+            for (var i = 0, len = bufferView.byteLength; i < len; i++)
                 binary += String.fromCharCode(bufferView[i]);
+            return binary;
```

Comment:
Why don't you think it's work base64 encoding the binary string? Any data that contains a lot of weird characters will pretty much explode in size because of the unicode encoding.